### ISSUE TITLE
IsFixedPitch data from post table lost on export

### ISSUE THREAD

**Issue by eigilnikolajsen** [opentypejs/opentype.js#611]
When I try to export my monospaced fonts using opentype.js they export with the post table value 'isFixedPitch' of 0 even though the value of the font object is 1 (1 meaning that it is a monospaced font, 0 meaning proportional).

## Expected Behavior
the 'isFixedPitch' value should follow the value of the font object's value

## Steps to Reproduce (for bugs)
Load [a monospaced font](https://fonts.google.com/specimen/DM+Mono) in OpenType.js and console.log the font object as well and the font.toTables(). Notice that the 'isFixedPitch' value in the post table is different in the font object and the tables.

## Your Environment
* Version used: 1.3.1
* Font used: [DM Mono](https://fonts.google.com/specimen/DM+Mono) but could be any monospaced font
* Browser Name and version: Chrome Version 115.0.5790.170 (Official Build) (x86_64)
* Operating System and version (desktop or mobile): Desktop
* Link to your project:

**Comment by brzzdev** [comment]
Any update here?

**Comment by Connum** [comment]
Nope, sorry. Nobody seems to have time to look into this currently. If someone finds where this needs to be set during export, a PR would be welcome.

**Comment by stlee42** [comment]
The title of this issue should be changed, because it is not just about the `post` table font metadata. Commit Mono's use case is about `isFixedPitch` flag in `CFF` table as well. My [comment](https://github.com/eigilnikolajsen/commit-mono/issues/15#issuecomment-1851042929) in that issue goes into it.

Currently, Commit Mono sets the flag correctly [here](https://github.com/eigilnikolajsen/commit-mono/blob/ff532ce10d0107bd3e82a9ea004b4c98f7215ae7/src/js/download_wizard.js#L323)
through here
https://github.com/opentypejs/opentype.js/blob/aa8ad76467a27f086a19a7d3e12479a3d4fd93aa/src/tables/sfnt.mjs#L347
but in here
https://github.com/opentypejs/opentype.js/blob/aa8ad76467a27f086a19a7d3e12479a3d4fd93aa/src/tables/cff.mjs#L1603
there is a bug in opentypejs after that line where `isFixedPitch` is ignored and not added to the `attrs` variable.

When it comes to `numberOfHMetrics`, opentypejs has a bug where it is not calculated, but instead set to the entire number of glyphs.
https://github.com/opentypejs/opentype.js/blob/aa8ad76467a27f086a19a7d3e12479a3d4fd93aa/src/tables/sfnt.mjs#L241
It should be set to the number of unique widths found in `hmtx` table.

Also, Commit Mono has a weird bug [here](https://github.com/eigilnikolajsen/commit-mono/blob/ff532ce10d0107bd3e82a9ea004b4c98f7215ae7/src/js/download_wizard.js#L320) where it sets it to 3, but where 1 is desired. That line can be erased, since the value ought to be calculated, not set. I think the code for setting fsSelection is wrong too when it comes to the regular style. ([bit 6 must be set](https://learn.microsoft.com/en-us/typography/opentype/spec/os2#fsselection))

This is just from reading the code, and I have not touched nor built any code. I beg your pardon that this is as far as I am inclined.

**Comment by stlee42** [comment]
There is a claim [here](https://github.com/eigilnikolajsen/commit-mono/issues/15#issuecomment-1790258981) that the original bug regarding the flag in the `post` table is already fixed in the master branch of opentypejs. So please consider this issue to be regarding the bugs mentioned in my previous post.
